tests/Functional/Connection/FetchTest.php

```diff
@@ -77,6 +79,41 @@ public function testFetchAllAssociative(): void
         ], $this->connection->fetchAllAssociative($this->query));
     }
 
+    public function testFetchAllKeyValue(): void
+    {
+        self::assertEquals([
```

Comment:
Depending on the answer about a defined behaviour of queries with more than 2 columns, tests need to be created to check that the methods behave as expected with more than two columns.

Reply:
This case is covered by `testFetchAllKeyValueWithLimit()`, and this is the sole reason why allow more than two columns. Do you think we should test this case more explicitly?

Reply:
If an explicit more than 2 columns are the same as with your DB2 use case, then there isn't any additional test necessary